src/dispatch/plugins/dispatch_core/plugin.py

```diff
@@ -165,6 +169,60 @@ def get_current_user(self, request: Request, **kwargs):
         return value
 
 
+class AwsAlbAuthProviderPlugin(AuthenticationProviderPlugin):
+    title = "Dispatch Plugin - AWS ALB Authentication Provider"
+    slug = "dispatch-auth-provider-aws-alb"
+    description = "AWS Application Load Balancer authentication provider."
+    version = dispatch_plugin.__version__
+
+    author = "ManyPets"
+    author_url = "https://manypets.com/"
+
+    @cached(cache=TTLCache(maxsize=1024, ttl=DISPATCH_AUTHENTICATION_PROVIDER_AWS_ALB_PUBLIC_KEY_CACHE_SECONDS))
+    def get_public_key(self, kid: str, region: str):
+        url = f"https://public-keys.auth.elb.{region}.amazonaws.com/{kid}"
+        req = requests.get(url)
+        return req.text
+
+    def get_current_user(self, request: Request, **kwargs):
+        credentials_exception = HTTPException(
+            status_code=HTTP_401_UNAUTHORIZED, detail=[{"msg": "Could not validate credentials"}]
+        )
+
+        encoded_jwt: str = request.headers.get('x-amzn-oidc-data')
+        if not encoded_jwt:
+            log.error(
+                f"Unable to authenticate. Header x-amzn-oidc-data not found."
+            )
+            raise credentials_exception
+
+        # Validate the signer
+        jwt_headers = encoded_jwt.split('.')[0]
+        decoded_jwt_headers = base64.b64decode(jwt_headers)
```

Comment:
This looks great! 🙌

QQ - Do we need to use `base64.urlsafe_b64decode` to decode JWT headers?